Library/ViewModels/SetYourPasswordViewModelTests.swift

```diff
@@ -1,33 +1,55 @@
-import XCTest
-
+@testable import KsApi
 @testable import Library
 import ReactiveExtensions_TestHelpers
+import XCTest
 
 final class SetYourPasswordViewModelTests: TestCase {
   private let viewModel = SetYourPasswordViewModel()
+
+  private let isLoading = TestObserver<Bool, Never>()
+  private let shouldShowActivityIndicator = TestObserver<Bool, Never>()
   private let saveButtonIsEnabled = TestObserver<Bool, Never>()
   private let contextLabelText = TestObserver<String, Never>()
   private let newPasswordLabel = TestObserver<String, Never>()
   private let confirmPasswordLabel = TestObserver<String, Never>()
+  private var setPasswordFailure = TestObserver<String, Never>()
+  private var setPasswordSuccess = TestObserver<Void, Never>()
+
+  private let setPasswordFailureService =
+    MockService(createPasswordResult: .failure(ErrorEnvelope(
+      errorMessages: ["Error creating password"],
+      ksrCode: nil,
+      httpCode: 1,
+      exception: nil
+    )))
+  private let setPasswordSuccessService =
+    MockService(createPasswordResult: .success(EmptyResponseEnvelope()))
 
   override func setUp() {
     super.setUp()
 
+    self.viewModel.outputs.shouldShowActivityIndicator.observe(self.shouldShowActivityIndicator.observer)
     self.viewModel.outputs.saveButtonIsEnabled.observe(self.saveButtonIsEnabled.observer)
     self.viewModel.outputs.contextLabelText.observe(self.contextLabelText.observer)
     self.viewModel.outputs.newPasswordLabel.observe(self.newPasswordLabel.observer)
     self.viewModel.outputs.confirmPasswordLabel.observe(self.confirmPasswordLabel.observer)
+    self.viewModel.outputs.setPasswordSuccess.observe(self.setPasswordSuccess.observer)
+    self.viewModel.outputs.setPasswordFailure.observe(self.setPasswordFailure.observer)
 
-    self.viewModel.inputs.configureWith("test@email.com")
     self.viewModel.inputs.viewDidLoad()
   }
 
   func test_init() {
-    self.contextLabelText
-      .assertValue("We will be discontinuing the ability to log in via Facebook. To log in to your account using the email test@email.com, please set a password that’s at least 6 characters long.")
-    self.newPasswordLabel.assertValue("Enter new password")
-    self.confirmPasswordLabel.assertValue("Re-enter new password")
-    XCTAssertNil(self.saveButtonIsEnabled.lastValue)
+    let userEnvelope = UserEnvelope(me: GraphUser.template)
+
+    withEnvironment(apiService: MockService(fetchGraphUserResult: .success(userEnvelope))) {
+      self.scheduler.advance()
+
```

Comment:
test the `contextLabelText` here and update the name of this test to describe what is being tested and the conditions.